source/extensions/filters/http/ext_proc/mutation_utils.cc

```diff
@@ -44,10 +44,72 @@ void MutationUtils::headersToProto(const Http::HeaderMap& headers_in,
   });
 }
 
-absl::Status MutationUtils::applyHeaderMutations(const HeaderMutation& mutation,
+absl::Status MutationUtils::responseHeaderSizeCheck(const uint32_t max_request_headers_kb,
+                                                    const uint32_t max_request_headers_count,
+                                                    const HeaderMutation& mutation,
+                                                    Counter& rejected_mutations) {
+  const auto& remove_headers = mutation.remove_headers();
+  if (remove_headers.size() > static_cast<int>(max_request_headers_count) ||
+      remove_headers.SpaceUsedExcludingSelf() > static_cast<int>(max_request_headers_kb * 1024)) {
```

Comment:
we don't need to add this check for remove.

To defend the large message, we need to set the gRPC channel limits.  This only work for google gRPC.  For Envoy gRPC, we need to figure out how to enforce that gRPC channel limits.